### ISSUE TITLE
Need help with the "MAKE IT MODULAR" module

### ISSUE THREAD

**Issue by haleyk10198** [nodeschool/discussions#1946]
It seems that the checker is blaming me for not including an early return, which I thought I have already included it.

Core code
```
var myFilter = require('./dirFilter.js');

myFilter(process.argv[2], process.argv[3], function(err, list){
	if(err)
		return console.log(err);
	for(var x in list)
		console.log(list[x]);
});
```

Module code
```
var fs = require('fs');

module.exports = function(path, extension, callback){
		fs.readdir(path, 'utf8', function(err, list){
			if(err)
			 	return callback(err);
			var re = new RegExp('^.+\.' + extension + '$');
			//var extension = /^.+\.#$/;
			//extension = extension.replace('#', process.argv[3]);
			//console.log(extension);
			var res = [];
			for(var x in list){
			 	if(list[x].search(re) != -1)
					res.push(list[x]);
			}
			callback(null, res);
	});
}
```

Checker result

 

> ✓
> 
>  Submission results match expected
> 
>  ✓
> 
>  Additional module file exports a single function
> 
>  ✓
> 
>  Additional module file exports a function that takes 3 arguments
> 
>  ✗
> 
>  Your additional module file [dirFilter.js] does not appear to pass back an
>  error received from fs.readdir(). Use the following idiomatic Node.js
>  pattern inside your callback to fs.readdir(): if (err) return
>  callback(err)

I wonder if I misinterpreting the concept of early return or did I implemented it incorrectly?

**Comment by drmason13** [comment]
Maybe @haleyk10198 has found a solution by now, but I managed to get your code working and there was only one small change needed.

**I removed the encoding argument from the call to readdir():**
`		fs.readdir(path, 'utf8', function(err, list){`
`		fs.readdir(path, function(err, list){`
and it passed the challenges.

I don't think this is intended behaviour for the workshop - My guess would be the workshop got muddled up by expecting the second argument to contain the function that handles errors in the "node way"
i.e. having err as first argument: function(err, data)

Your optional argument of encoding seems perfectly fine to me having checked the [documentation for readdir](https://nodejs.org/api/fs.html#fs_fs_readdir_path_options_callback) but isn't needed since the default is 'utf8' anyway.

readdir treats the filenames as strings by default, so removing the encoding hasn't changed anything. Good news is, your code is fine! :)
